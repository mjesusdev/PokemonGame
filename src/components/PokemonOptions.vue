<template>
  <div class="options-container">
    <ul class="list-group d-grid gap-2">
      <li class="list-group-item btn mb-2" v-for="pokemon in pokemons" 
        :key="pokemon.id"
        @click="$emit( 'selection-pokemon', pokemon.id )">
          {{ pokemon.name }}
      </li>
    </ul>
  </div>
</template>

<script>
  export default {
    props: {
      pokemons: {
        type: Array,
        required: true,
      },
    }
  }
</script>

<style scoped>
  /* Pokemon Options */
  li {
      background: #f4a261 !important;
      border-top-width: thin !important;
      font-weight: bold;
      cursor: pointer;
      width: 25em;
  }

  li:hover {
      background: #b67d4f !important;
  }

  .options-container {
      font-family: Arial, Helvetica, sans-serif !important;
      display: flex;
      justify-content: center;
  }

  @media (max-width: 768px) {
    li {
      width: 15em;
    }
  }
</style>