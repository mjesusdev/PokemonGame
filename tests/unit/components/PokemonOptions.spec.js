import { shallowMount } from '@vue/test-utils'
import PokemonOptions from '@/components/PokemonOptions.vue'

import { pokemons } from '../mocks/pokemons.mock'

describe('PokemonOptions Component', () => {

    let wrapper

    beforeEach(() => {
        wrapper = shallowMount(PokemonOptions, {
            props: {
                pokemons
            }
        })
    })

    test('should do match with Snapshot', () => {
        expect(wrapper.html() ).toMatchSnapshot()
    })

    test('should show the 4 options correctly', () => {
        const liTags = wrapper.findAll('li')
        expect( liTags.length ).toBe(4)

        expect( liTags[0].text() ).toBe('pikachu');
        expect( liTags[1].text() ).toBe('charmander');
        expect( liTags[2].text() ).toBe('venusaur');
        expect( liTags[3].text() ).toBe('mew');
    })

    test('should emit the correct pokemon when clicked', () => {
        const [ li1, li2, li3, li4 ] = wrapper.findAll('li')

        li1.trigger('click')
        li2.trigger('click')
        li3.trigger('click')
        li4.trigger('click')

        expect( wrapper.emitted('selection-pokemon').length ).toBe(4);
        expect( wrapper.emitted('selection-pokemon')[0] ).toStrictEqual([5]);
        expect( wrapper.emitted('selection-pokemon')[1] ).toStrictEqual([10]);
        expect( wrapper.emitted('selection-pokemon')[2] ).toStrictEqual([15]);
        expect( wrapper.emitted('selection-pokemon')[3] ).toStrictEqual([20]);
    })

})