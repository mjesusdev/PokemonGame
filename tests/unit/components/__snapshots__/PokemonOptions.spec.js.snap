// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component should do match with Snapshot 1`] = `
<div class="options-container">
  <ul class="list-group">
    <li class="list-group-item mb-2">pikachu</li>
    <li class="list-group-item mb-2">charmander</li>
    <li class="list-group-item mb-2">venusaur</li>
    <li class="list-group-item mb-2">mew</li>
  </ul>
</div>
`;
